dotnet/src/Agents/Core/ChatCompletionAgent.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// A <see cref="KernelAgent"/> specialization based on <see cref="IChatCompletionService"/>.
+/// </summary>
+public sealed class ChatCompletionAgent : ChatHistoryKernelAgent
+{
+    /// <inheritdoc/>
+    public override string? Description { get; }
+
+    /// <inheritdoc/>
+    public override string Id { get; }
+
+    /// <inheritdoc/>
+    public override string? Name { get; }
+
+    /// <summary>
+    /// Optional execution settings for the agent.
+    /// </summary>
+    public PromptExecutionSettings? ExecutionSettings { get; set; }
+
+    /// <inheritdoc/>
+    public override async IAsyncEnumerable<ChatMessageContent> InvokeAsync(
+        IReadOnlyList<ChatMessageContent> history,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var chatCompletionService = this.Kernel.GetRequiredService<IChatCompletionService>();
+
+        ChatHistory chat = new();
+        if (!string.IsNullOrWhiteSpace(this.Instructions))
+        {
+            chat.Add(new ChatMessageContent(AuthorRole.System, this.Instructions) { AuthorName = this.Name });
+        }
+        chat.AddRange(history);
```

Comment:
Are we allowing someone to pass in a `history` that contains a new system message?

Elsewhere I think we disallow this and throw an exception.